lib/ash_json_api/controllers/helpers.ex

```diff
@@ -511,7 +511,15 @@ defmodule AshJsonApi.Controllers.Helpers do
   end
 
   defp fields(request, resource) do
-    Map.get(request.fields, resource) || request.route.default_fields || []
+    fields = Map.get(request.fields, resource) || request.route.default_fields || []
+    field_inputs = request.field_inputs[resource] || %{}
+
+    Enum.map(fields, fn field ->
+      case Map.get(field_inputs, field) do
+        nil -> field
+        value -> {field, value}
```

Comment:
we may want to check if its a calculation here, and if not show an invalid error, to avoid trying to provide inputs when things don't have inputs.

Reply:
ah, I see that happens already somewhere else, nvm 👍